### PR TITLE
[FEAT] Switch to V2 endpoints for some endpoints (#1281)

### PR DESCRIPTION
**Pull request by Mildophin**
### Description

Switch to V2 endpoints for the following endpoints:
- PUT cancel jobs
- POST add jobs
- PUT cancel batch
- PUT complete batch
- PUT cancel job

The V2 remove the 'results' field in the response, field present in the V1 of these endpoints.

### Checklist

- [x] The title of the PR follows the right format: [{Label}] {Short Message}. Label examples: IMPROVEMENT, FIX, REFACTORING... Short message is about what your PR changes.

#### Versioning (PASQAL developers only)

- [x] Update the version of pasqal-cloud in `_version.py` following the changes in your PR and by using [semantic versioning](https://semver.org/).

#### Documentation

- [x] Update CHANGELOG.md with a description explaining briefly the changes to the users.

#### Tests

- [x] Unit tests have been added or adjusted.
- [ ] Tests were run locally.

#### Internal tests pipeline (PASQAL developers only)
- [ ] Update and run the internal tests while targeting the branch of this PR.
 If your PR hasn't changed any functionality, it still needs to be validated against internal tests.

#### After updating the version (PASQAL developers only)

- [ ] Open a PR on the internal tests that updates the version used for the pasqal-cloud backward compatibility tests.

### CODE REVIEW

**Comment by Augustinio** [comment]
Technically if we remove a key from the response, doesn't it constitute a breaking change? If so the minor should be bumped.

**Comment by Mildophin** [comment]
> Technically if we remove a key from the response, doesn't it constitute a breaking change? If so the minor should be bumped.

Indeed it is, just realised it, I will bump it to 0.13.0

**Review by Augustinio — APPROVED** [review]

**Review by nmenetrier — APPROVED** [review]

**Review by MohamedAminMallek — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,25 +2,36 @@
...
+## [0.13.0] - 2024-09-04
+
+### Breaking change
+
+- The following methods now use the V2 that do not return the results any more:
+    - Cancel a batch
+    - Cancel a job
+    - Cancel a group of jobs
+    - Add jobs to a batch
+    - Close a batch
```
[the remark this answers is not in the document]
Comment (Mildophin) [reply CHANGELOG.md:14]: Amin is actually right as we have a property that set the result in the job Class, so the user will not notice any difference, I bring back it to 0.12.1 and a different changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,25 +2,36 @@
 
 All notable changes to this project will be documented in this file.
 
+## [0.12.1] - 2024-09-10
+
+### Changed
+
+- Better performances for the following methods:
```
Comment (oliver-gordon) [inline CHANGELOG.md:9]: Nit: There is no performance gain in this SDK. You have just updated the endpoints to v2.

**Review by oliver-gordon — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Mildophin) [reply CHANGELOG.md:9]: However the user does not see it, it does 0 difference for it, should I write it down instead ?

**Pull request merged**